src/libcsg/modules/io/xmltopologyreader.cc

```diff
@@ -134,25 +136,58 @@ void XMLTopologyReader::ParseMolecule(Property &p, string molname, int nbeads, i
             } catch (runtime_error) {
                 atq = 0.0;
             }
+            try {
+                resid = it->getAttribute<int>("resid");
+                if (resid<=0)
+                    throw std::runtime_error(
```

Comment:
This runtime_error will be catched by the "catch" below and the code will proceed with resid=-1; Is it an intention?